planner/core/common_plans.go

```diff
@@ -374,7 +374,7 @@ REBUILD:
 	e.names = names
 	e.Plan = p
 	_, isTableDual := p.(*PhysicalTableDual)
-	if !isTableDual && prepared.UseCache {
+	if !isTableDual && prepared.UseCache && !stmtCtx.OptimDependOnMutableConst {
```

Comment:
Nice. It seems all places that use `ContainMutableConst` should be modified. We can only modify `EvaluateExprWithNull` in this PR and change all other places later in another PR.